src/model/base_model.py

```diff
@@ -0,0 +1,82 @@
+import tensorflow as tf
+from tensorflow.keras import layers
+
+
+def fill_fc_weights(fc):
+    if isinstance(fc, tf.keras.Sequential):
+        all_layers = fc.layers
+    else:
+        all_layers = fc
+
+    for m in all_layers:
+        if isinstance(m, layers.Conv2D):
+            if m.use_bias:
+                m.bias_initialier = tf.keras.initializers.constant(0)
+
+
+class BaseModel(tf.keras.Model):
+    def __init__(self, heads, head_convs, num_stacks, opt=None):
+        super().__init__()
+        if opt is not None and opt.head_kernel != 3:
+            print('Using head kernel:', opt.head_kernel)
+            head_kernel = opt.head_kernel
+        else:
+            head_kernel = 3
+        self.num_stacks = num_stacks
+        self.heads = heads
+        for head in self.heads:
+            classes = self.heads[head]
+            head_conv = head_convs[head]
+            if len(head_conv) > 0:
+                out = layers.Conv2D(classes,
+                                    kernel_size=1, strides=1, padding='valid', use_bias=True,
+                                    data_format='channels_first')
+                conv = layers.Conv2D(head_conv[0], kernel_size=head_kernel, padding='same', use_bias=True,
+                                     data_format='channels_first')
+                convs = [conv, layers.ReLU()]
+                for k in range(1, len(head_conv)):
+                    convs.extend(
+                            [layers.Conv2D(head_conv[k], kernel_size=1, use_bias=True, data_format='channels_first'),
+                             layers.ReLU()])
+                convs.append(out)
+                fc = tf.keras.Sequential(convs)
+                if 'hm' in head:
+                    fc.layers[-1].bias_initizlier = tf.keras.initializers.constant(opt.prior_bias)
+                else:
+                    fill_fc_weights(fc)
+                    pass
+            else:
+                fc = layers.Conv2D(classes, kernel_size=1, strides=1, padding='valid', use_bias=True)
+                if 'hm' in head:
+                    fc.bias.data.fill_(opt.prior_bias)
+                else:
+                    fill_fc_weights(fc)
+                    pass
+            self.__setattr__(head, fc)
```

Comment:
Why are you using `__setattr__` instead of an assigment?

Reply:
the name of the instance attribute is given by head (a string), and the value by fc (a keras layer or model).
Basically, the name of the layers and their parameters are given in the parameters.